### PR TITLE
docs: fix v2 redirects

### PR DESCRIPTION
**Pull request by trevorblades**
This PR removes the `/federation-2/*` redirect and removes the `/v2` path prefix infront of all v2 redirects.

### CODE REVIEW

File: docs/source/_redirects [hunk 1]
```diff
@@ -1,15 +1,13 @@
-/federation-2/*  /docs/federation/v2/federation-2/:splat
+/federation-2/coming-to-federation-2/ /docs/federation/v2/federation-2/under-consideration/
```
Comment (StephenBarlow) [inline docs/source/_redirects:1]: We can actually cut these first two (I've done so in a separate PR that hasn't landed yet)

**Review by StephenBarlow — APPROVED** [review]
Approved with one optional note 🤷

File: docs/source/_redirects (hunk 1 above)
Comment (trevorblades) [reply docs/source/_redirects:1]: Let's just let your PR remove them separately for hopefully fewer merge conflicts.
